pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -73,13 +75,19 @@ func forThisNodePredicate(cl client.Client) predicate.Funcs {
 			return nodeSelectorMatchesThisNode(cl, createEvent.Object)
 		},
 		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
-			return nodeSelectorMatchesThisNode(cl, deleteEvent.Object)
+			return false
 		},
 		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
-			return nodeSelectorMatchesThisNode(cl, updateEvent.ObjectNew)
-		},
-		GenericFunc: func(genericEvent event.GenericEvent) bool {
-			return nodeSelectorMatchesThisNode(cl, genericEvent.Object)
+			if !nodeSelectorMatchesThisNode(cl, updateEvent.ObjectNew) {
```

Comment:
We are missing a lot of stuff from https://github.com/nmstate/kubernetes-nmstate/blob/master/pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go#L70-L94

Reply:
The reason is that with the new design we reapply re-reconcile the configuration periodically, the check whether we need to Update or not should haven via the dry-run we discussed. We need to compare the desired configuration with the state of host, not with the previous desired configuration.

Reply:
Ahh that's right, ok.